### PR TITLE
[OBM-2818]MobileApp Firebase notifications

### PR DESCRIPTION
**Pull request by VictorKhoroshko**
**Why:** Needs an ability to send mobile app push notifications.
**How:** Firebase service implemented. 
It uses configuration from Configuration ms and supports proxy settings. For interactions with Google FCM API uses Java Firebase admin SDK.

### CODE REVIEW

File: src/main/java/com/icthh/xm/tmf/ms/communication/lep/XmCommunicationMsLepProcessingApplicationListener.java [hunk 1]
```diff
@@ -1,34 +1,53 @@
...
 
     private final BusinessTimeConfigService tenantConfigService;
     private final RestTemplate restTemplate;
+    private final RestTemplate loadBalancedRestTemplate;
     private final CommonsService commonsService;
     private final PermissionCheckService permissionCheckService;
     private final KafkaTemplateService kafkaTemplateService;
 
+    public XmCommunicationMsLepProcessingApplicationListener(BusinessTimeConfigService tenantConfigService,
+                                                             RestTemplate restTemplate,
```
Comment (igordumchykov) [inline src/main/java/com/icthh/xm/tmf/ms/communication/lep/XmCommunicationMsLepProcessingApplicationListener.java:37]: should you add qualifier here?

Comment (sergeysenja1992) [reply src/main/java/com/icthh/xm/tmf/ms/communication/lep/XmCommunicationMsLepProcessingApplicationListener.java:37]: i this it's correct, it's rest template for external call. next parameter it's rest template for internal call between ms

Comment (sergeysenja1992) [reply src/main/java/com/icthh/xm/tmf/ms/communication/lep/XmCommunicationMsLepProcessingApplicationListener.java:37]: and for me better to configure lombok than add constructor only for annotation
https://projectlombok.org/features/constructor

Comment (VictorKhoroshko) [reply src/main/java/com/icthh/xm/tmf/ms/communication/lep/XmCommunicationMsLepProcessingApplicationListener.java:37]: @SergeyPentsov Indeed, an explicit ctor is redundant, removed.

**Review by igordumchykov — APPROVED** [review]

**Review by pkev777 — APPROVED** [review]

**Pull request merged**